test/context.js

```diff
@@ -89,6 +90,58 @@ describe('context', function () {
       .done();
   });
 
+  function saveInstanceArray(instanceArray) {
+    return helper.dal.context.save(instanceArray.pop())
+    .then(function() {
+      if (instanceArray.length > 0) {
+        return saveInstanceArray(instanceArray);
+      }
+    });
+  }
+
+  function resultEquivalentExpectations(result, expectedItems) {
+    if (result.length !== expectedItems.length) {
+      return false;
+    }
+
+    var x;
+    for (x in result) {
```

Comment:
Just use result.forEach(function(item) {}) instead of having to declare a var for x.